HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1359,6 +1473,123 @@ def self.apply_infiltration_and_ventilation_fans(model, weather, vent_mech, vent
+    if not vent_mech_erv_hrv.empty?
+      # Actuators for ERV/HRV
+      sens_name = "#{Constants.ObjectNameERVHRV} sensible load"
+      erv_sens_load_var = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, sens_name.gsub(' ', '_'))
+      erv_sens_load_actuator = create_sens_lat_load_actuator_and_equipment(model, sens_name, @living_space, 0.0, 0.0)
+      lat_name = "#{Constants.ObjectNameERVHRV} latent load"
+      erv_lat_load_var = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, lat_name.gsub(' ', '_'))
+      erv_lat_load_actuator = create_sens_lat_load_actuator_and_equipment(model, lat_name, @living_space, 1.0, 0.0)
+
+      infil_program.addLine("Set #{erv_sens_load_actuator.name} = 0.0")
+      infil_program.addLine("Set #{erv_lat_load_actuator.name} = 0.0")
+
+      # ERV/HRV EMS load model
+      # E+ ERV model is using standard density for MFR calculation, caused discrepancy with other system types.
+      # E+ ERV model also does not meet setpoint perfectly.
+      # Therefore ERV is modeled within EMS infiltration program
+      # Air property at inlet nodes in two sides of ERV
+
+      # Common part across multiple erv/hrv programs, create only once
+      infil_program.addLine("Set ERVSupInPb = #{@pbar_sensor.name}") # oa barometric pressure
+      infil_program.addLine("Set ERVSupInTemp = #{@tout_sensor.name}") # oa db temperature
+      infil_program.addLine("Set ERVSupInW = #{@wout_sensor.name}")   # oa humidity ratio
+      infil_program.addLine('Set ERVSupRho = (@RhoAirFnPbTdbW ERVSupInPb ERVSupInTemp ERVSupInW)')
+      infil_program.addLine('Set ERVSupCp = (@CpAirFnW ERVSupInW)')
+      infil_program.addLine('Set ERVSupInEnth = (@HFnTdbW ERVSupInTemp ERVSupInW)')
+
+      infil_program.addLine("Set ERVSecInTemp = #{@tin_sensor.name}") # zone air temperature
+      infil_program.addLine("Set ERVSecInW = #{@win_sensor.name}") # zone air humidity ratio
+      infil_program.addLine('Set ERVSecCp = (@CpAirFnW ERVSecInW)')
+      infil_program.addLine('Set ERVSecInEnth = (@HFnTdbW ERVSecInTemp ERVSecInW)')
+    end
+
+    # Calculate cfm weighted average effectiveness for hvac sizing
+    # Please review this, and its integration with hvac_sizing
+    weighted_vent_mech_tot_eff = 0.0
+    weighted_vent_mech_lat_eff = 0.0
+    weighted_vent_mech_apparent_sens_eff = 0.0
+
+    # Apply ERV/HRV mechanical ventilation
+    vent_mech_erv_hrv.each do |vent_mech|
+      vent_mech_cfm = vent_mech.flow_rate * (vent_mech.hours_in_operation / 24.0)
+      vent_mech_fan_w = vent_mech.fan_power * (vent_mech.hours_in_operation / 24.0)
+      vent_mech_sens_eff, vent_mech_lat_eff, vent_mech_apparent_sens_eff = calc_hrv_erv_effectiveness(vent_mech, vent_mech_cfm, vent_mech_fan_w)
+      # No need to add balanced system here to average because their effectivenesses are all 0.0
+      weighted_vent_mech_tot_eff += vent_mech_cfm / tot_bal_cfm * vent_mech.total_recovery_efficiency.to_f
```

Comment:
I assume this works correctly if the HPXML input is AdjustedTotalRecoveryEfficiency instead of TotalRecoveryEfficiency? I can't immediately tell.

Reply:
This line was deleted based on above comments.